### DISCUSSION TITLE
Orca not accepting standard Klipper/G-Code

### DISCUSSION THREAD

**Discussion by stratiegery01** [OrcaSlicer/OrcaSlicer#1142] [Q&A] [answered]
After using SuperSlicer for a few years working fine, my machine start G-code doesnt transfer over to Orca nicely even though the settings are set for Klipper as best I can tell. The problem I'm having is on line 5, about 'first layer height'. Attempting to slice this brings up a parsing error which I cant seem to figure out, claiming this line is "invalid g-code". Thoughts?

Here's the old code (which definitely works on the same machine) from Superslicer:
m140 s{first_layer_bed_temperature} ; bed temp warm
m109 s{first_layer_temperature} ; nozzle temp and wait
m190 s{first_layer_bed_temperature} ; bed temp and wait
g92 e0
g1 x0 y0 z{first_layer_height} f3000 ; wipe start location
g1 x75 e9 f1000 ; underextrude wipe
g1 x150 e12.5 f1000 ; standard wipe
g1 z5 e-5 f200 ; filament retraction
g92 e0
g90

*I also tried the pre-loaded machine G-codes that are added by default when configuring a new machine in Orca, and even those didnt work.

**Comment by Alex00101** [comment]
Parameter name I think is initial_layer_print_height

**Comment by SoftFever** [comment, marked as answer]
@stratiegery01 
There are syntax errors in your start g-code
1. `first_layer_temperature` and `first_layer_temperature` are vectors, so you will need [] to access them
Or use this syntax `m140 s{first_layer_bed_temperature[0]} ; bed temp warm`
2. Orca Slicer use `initial_layer_print_height` instead of `first_layer_height` variable name. 

```
m140 s[first_layer_bed_temperature] ; bed temp warm
m109 s[first_layer_temperature] ; nozzle temp and wait
m190 s[first_layer_bed_temperature] ; bed temp and wait
g92 e0
g1 x0 y0 z{initial_layer_print_height} f3000 ; wipe start location
g1 x75 e9 f1000 ; underextrude wipe
g1 x150 e12.5 f1000 ; standard wipe
g1 z5 e-5 f200 ; filament retraction
g92 e0
g90
```

**Reply by stratiegery01** [reply]
where does one find what parameter names Orca Slicer uses compared to other platforms?

I'm suprised that the slicer is not able to follow the standard list/syntax from Klipper: https://www.klipper3d.org/G-Codes.html#temperature_wait

**Reply by SoftFever** [reply]
It seems that you may have confused it with other things. 
Please note that all PrusaSlicer-derived slicers including OrcadSlicer follow the **STANDARD** syntax as described in this link: https://help.prusa3d.com/article/macros_1775

**Reply by SoftFever** [reply]
OrcaSlicer does not currently have a compiled list of placeholders. However, it supports the placeholders listed here: [placeholder](https://wiki.bambulab.com/en/software/bambu-studio/placeholder-list). Additionally, OrcaSlicer is reintroducing PrusaSlicer placeholders for improved compatibility. Currently, most common placeholders are supported, including the first_layer_height variable mentioned in this discussion (only available in the develop branch at the moment).

**Reply by stratiegery01** [reply]
Thank you @SoftFever! Using these placeholders seemed to solve the issue I was having. 

After slicing and starting a print from OrcaSlicer, I still do see in the Klipper console though "Unknown command:"EXCLUDE_OBJECT_DEFINE", Unknown command:"EXCLUDE_OBJECT_START", and "Unknown command:"EXCLUDE_OBJECT_END" errors; though I know I am not the only one seeing this problem and deserves a separate discussion on.

**Reply by SoftFever** [reply]
@stratiegery01 
Please refer to this part in the readme file: https://github.com/SoftFever/OrcaSlicer#note
You will need to enable Enable object exclusion in Klipper to use this feature.
Alternatively, you can turn off the feature in OS: 
![image](https://github.com/SoftFever/OrcaSlicer/assets/103989404/08b34ba2-aef2-4f76-a684-749a5fa02ce5)

It's a very nice and handy feature though, OrcaSlicer has supported it natively, it's recommended to enable it in Klipper.

**Comment by TechnikusMR** [comment]
@SoftFever 

Thank you for your Solution.
